docs/api/conventions.md

```diff
@@ -91,29 +91,147 @@ There are a few conventions that Remix uses you should be aware of.
 
-  Params can be nested routes, just create a folder with the `$` in it.
+#### Dot Delimeters
 
-- **`app/routes/files/$.tsx`**: To add a "splat" path (some people call this a "catchall") name the file simply `$.tsx`. It will create a route path pattern like `files/*`. You can also use this along with dot file names: `app/routes/files.$.tsx`.
+```[8]
+app
+| routes
+| - blog
+| --- blog.tsx
+| --- categories.tsx
+| --- index.tsx
+| - about.tsx
+| - blog.authors.tsx
+| - index.tsx
+| root.tsx
+```
+
+By creating a file with `.` characters between segments, you can create a nested URL without nested layouts. For example, a file `/app/routes/blog.authors.tsx` will route to the pathname `/blog/authors`, but it will not share a layout with routes in the `/app/routes/blog/` directory.
+
+#### Dynamic Route Parameters
+
+```[3]
+app
+| routes
+| - blog
+| --- $postId.tsx
+| --- blog.tsx
+| --- categories.tsx
+| --- index.tsx
+| - about.tsx
+| - blog.authors.tsx
+| - index.tsx
+| root.tsx
+```
+
+Routes that begin with a `$` character indicate the name of a dynamic segment of the URL. It will be parsed and passed to your loader and action data as a value on the `param` object.
+
+For example: `/app/routes/blog/$postId.tsx` will match the following URLs:
+
+- `/blog/my-story`
+- `/blog/once-upon-a-time`
+- `/blog/how-to-ride-a-bike`
+
+On each of these pages, the dynamic segment of the URL path is the value of the parameter in your route's data.
+
+Also noted: nested routes can contain dynamic segments by using the `$` character in the parent's directory name. For example, `/app/routes/blog/$postId/edit.tsx` might represent the editor view for blog entries.
+
+See the <Link to="../routing">routing guide</Link> for more information.
+
+<docs-info>Some CLIs require you to escape the \$ when creating files:
+
+```bash
+touch routes/\$params.tsx
+```
+
+</docs-info>
+
+#### Splat Routes
+
+```[4,9]
+app
+| routes
+| - blog
+| --- $.tsx
+| --- $postId.tsx
+| --- blog.tsx
+| --- categories.tsx
+| --- index.tsx
+| - $.tsx
+| - about.tsx
+| - blog.authors.tsx
+| - index.tsx
+| root.tsx
+```
 
-- **`app/routes/__some-layout/some-path.tsx`**: Prefixing a folder with `__` will create a "layout route". Layout routes are routes that don't add anything to the URL for matching, but do add nested components in the tree for layouts. Make sure to also have `__some-layout.tsx` as well. For example, all of your marketing pages could share a layout in the route tree with `app/routes/__marketing.tsx` as the layout and then all of the child routes go in `app/routes/__marketing/products.tsx` and `app/routes/__marketing/buy.tsx`. The `__marketing.tsx` route won't add any segments to the URL, but it will render when it's child routes match.
+Files that are named `$.tsx` are called "splat" (or "catch-all") routes. These routes will map to any URL not matched by other route files in the same directory.
```

Comment:
Would probably be good to expand this a bit to explain how to access the splat value (I actually don't know how myself 😅).